avm/src/daemon/ctrl.rs

```diff
@@ -400,4 +586,169 @@ impl ControlPort {
       }
     }
   }
+
+  pub fn get_vm_for_key(self: &ControlPort, key: &str) -> &VMMetadata {
+    &self.vms[self.lrh.get_primary_node_id(key)]
+  }
```

Comment:
should we have this return `Option<&VMMetadata>` and do `self.vms.get(self.lrh.get_primary_node_id(key)`?

Reply:
This should be an infallible path, though. `get_primary_node_id` should be returning an array index within the range of the `vms` since that `vms` array is literally what the LogRendezvousHash logic is indexing.